docs/developer-guide/code-conventions.md

```diff
@@ -0,0 +1,46 @@
+# Code conventions
+In order to preserve quality, maintainability and testability when you develop in MapStore 2  you should follow the following rules and best practices. 
+
+# TL;DR
+ - Access to the state using state selectors
+ - Prefer `initialState` for initial/default configuration over plugin configuration
+ - Reuse components with HOCs or create stateless and context-less new components (as much as possible)
+ - Use axios and redux-observable for async
+ 
+## Access to the state using state selectors
+Is **strongly recommended** to not access to the state directly inside the `mapStateToProps` function of `react-redux` .
+Use (or define) selectors in the `selectors` directory. This provides the following advantages:
+ - Selectors can be reused in epics.
+ - Ready to use optimization with reselect
+ - Simplify future refactoring
+ - Easy unit testing and bug identification
+ 
+Wrapping all the access to the state inside well-defined selector makes easier to add functionalities and will increase code maintainability. You should always reuse existing selectors (or create new ones) to access to the application state for core application functionalities.
+It will help also future refactoring because any change to the state structure (from the reducer point of view) or data source (from the components point of view) requires only changes to the interested selectors.
+
+A selector should be placed into the proper `selectors/<state-slice>.js` file with the same name of the relative reducer.
+When a selector retrieves data from more than one state slices, you should place it in the selector nearest by concern. For instance `isFeatureGridOpen` should be placed into `featuregrid`
+
+If you don't work on a core functionality, where the state is shared between many components, defining the selector directly in the plug-in is not denied (e.g. examples).
+
+## Prefer `initialState` over plugin configuration
+In the past we had a plug-in configuration and initial state. The plug-in configuration is now considered a bad practice. Use the `initialState` to configure the application.
```

Comment:
I don't agree, you cannot put everything in the initial state, only stuff that is connected through redux, configuration is still a good practice for properties that are not connected to the state (elsewhere, how can you configure them?)

Reply:
Good point. Can you specify that?